resources/user.go

```diff
@@ -59,8 +63,31 @@ func (obj *UserRes) Default() Res {
 
 // Validate if the params passed in are valid data.
 func (obj *UserRes) Validate() error {
+	const badChars string = " \"{}()<>&*‘|=?;[]$–#~!.”%%/\\:+,`"
+
 	if obj.State != "exists" && obj.State != "absent" {
-		return fmt.Errorf("State must be 'exists' or 'absent'")
+		return fmt.Errorf("state must be 'exists' or 'absent'")
+	}
+	if obj.GID != nil && obj.Group != nil {
+		return fmt.Errorf("cannot use both GID and Group")
+	}
+	if obj.Group != nil {
+		if strings.ContainsAny(*obj.Group, badChars) {
+			return fmt.Errorf("group contains invalid character(s)")
+		}
+		if *obj.Group == "" {
+			return fmt.Errorf("group cannot be empty string")
+		}
+	}
+	if obj.Groups != nil {
+		for _, group := range obj.Groups {
+			if strings.ContainsAny(group, badChars) {
+				return fmt.Errorf("groups list contains invalid character(s)")
+			}
+			if group == "" {
```

Comment:
nit: do the cheap test first, also makes it easier to read in case user is not sure what happens if you pass in an empty string...